src/Connections.Abstractions/Features/IGracefulConnectionLifetimeFeature.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Connections.Features
+{
+    public interface IGracefulConnectionLifetimeFeature
```

Comment:
Any naming suggestions?

Reply:
```csharp
public interface IConnectionLifetimeNotificationFeature
{
    CancellationToken ConnectionCloseRequested { get; set; }
    void RequestClose();
}
```

Reply:
Hmm, not bad.  ConnectionCloseRequested -> ConnectionClosing?

Reply:
RequestClose -> Close?

Reply:
Close would definitately work as its parallel is `Abort` so its similar but less severe 
```csharp
public interface IConnectionLifetimeFeature
{
    CancellationToken ConnectionClosed { get; set; }
    void Abort();
}
```

Just wondering how the token use would look in code

```csharp
ConnectionClosingGracefully.Register( ... );

if (ConnectionClosingGracefully.IsCancellationRequested)
```

```csharp
ConnectionCloseRequested.Register( ... );

if (ConnectionCloseRequested.IsCancellationRequested)
```

```csharp
ConnectionClosing.Register( ... );

if (ConnectionClosing.IsCancellationRequested)
```